### PR TITLE
Minnow config

### PR DESCRIPTION
**Pull request by PLangowski**

### CODE REVIEW

File: platform-configs/minnowboard-turbot.robot [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Flash Mbt
+    [Documentation]    Flash Device Under Test firmware, check flashing result
+    ...    and set RTE relay to OFF state. Implementation must be
+    ...    compatible with the theory of operation of a specific
+    ...    platform.
+    Sleep    1s
+    Power Cycle Off
+    IF    '${FLASH_OPT}'=='full'
+        ${flash_result}    ${rc}=    SSHLibrary.Execute Command
+        ...    flashrom -f -p linux_spi:dev=/dev/spidev1.0,spispeed=8000 -w /tmp/coreboot.rom 2>&1
```
Comment (miczyg1) [inline platform-configs/minnowboard-turbot.robot:57]: The flashrom commands will need the chip parameter: `-c W25Q...` IIRC

File: dasharo-compatibility/basic-interfaces.robot [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keyword
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+BI001.001 Check serial output
```
Comment (miczyg1) [inline dasharo-compatibility/basic-interfaces.robot:29]: Cannot we simply do an Enter TianoCore Setup test?

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: dasharo-compatibility/basic-interfaces.robot (hunk 2 above)
Comment (macpijan) [reply dasharo-compatibility/basic-interfaces.robot:29]: We do not have Dasharo firmware here (yet).

I proposed that we could also have a test checking operation of control interfaces, despite the firmware/payload type.

We could use such one to verify that platform was setup correctly.

**Comment by macpijan** [comment]
@PLangowski please rebase and change target to develpo

File: dasharo-compatibility/basic-interfaces.robot (hunk 2 above)
Comment (miczyg1) [reply dasharo-compatibility/basic-interfaces.robot:29]: Yes, but the official firmware is TianoCore based...

Comment (macpijan) [reply dasharo-compatibility/basic-interfaces.robot:29]: If this is the case here, maybe we can see if the `enter boot menu` tests pass as well on this platform.
I am not sure, though, if having this basic generic test at the same time is a problem.

File: platform-configs/minnowboard-turbot.robot (hunk 1 above)
Comment (PLangowski) [reply platform-configs/minnowboard-turbot.robot]: added aa45de8b2cb3911b75a47249711f6df3f15f602b

**Comment by PLangowski** [comment]
@macpijan I rebased the branch

File: dasharo-compatibility/basic-interfaces.robot (hunk 2 above)
Comment (PLangowski) [reply dasharo-compatibility/basic-interfaces.robot:29]: I downloaded the firmware from [intel.com](https://www.intel.com/content/www/us/en/download/674533/minnowboard-max-and-minnowboard-turbot-firmware-flash-images.html?wapkw=minnowboard%20turbot) and I can't enter setup. Instead, a lot of logs are printed, including this:
```
[Bds]=============Begin Load Options Dumping ...=============
  Driver Options:
  SysPrep Options:
  Boot Options:
    Boot0000: UEFI BootManagerMenuApp            0x0109
    Boot0001: Internal UEFI Shell 2.0            0x0001
    Boot0002: Enter Setup                0x0109
    Boot0003: UEFI UiApp                 0x0001
    Boot0004: UEFI PXEv4 (MAC:0008A20CB8BF)              0x0001
    Boot0005: UEFI PXEv6 (MAC:0008A20CB8BF)              0x0001
    Boot0006: UMnpSyncSendPacket: No network cable detected.
EFI HTTPv4 (MAC:0008A20CB8BF)            0x0001
    Boot0007: UEFI HTTPv6 (MAC:0008A20CB8BF)             0x0001
  PlatformRecovery Options:
    PlatformRecovery0000: Default PlatformRecovery               0x0001
[Bds]=============End Load Options Dumping=============
[Bds]BdsWait ...Zzzzzzzzzzzz...
[Bds]BdsWait(5)..Zzzz...
[Bds]BdsWait(4)..Zzzz...
[Bds]BdsWait(3)..Zzzz...
[Bds]BdsWait(2)..Zzzz...
[Bds]BdsWait(1)..Zzzz...
[Bds]Exit the waiting!
```
 After that I'm thrown into UEFI shell. I can type `exit`, which lets me choose the boot option, but when I choose `UEFI UIApp` I end up back in UEFI shell with a `map: No mapping found.` message.
So entering Tianocore setup isn't something I can do right now.

**Comment by macpijan** [comment]
@PLangowski `robotidy` modifies files in this PR, which suggests you may not have pre-commit installed locally? You may also try running on all files: `pre-commit run --all-files`.

File: platform-configs/minnowboard-turbot.robot [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    RteCtrl Power On
+
+Flash Mbt
+    [Documentation]    Flash Device Under Test firmware, check flashing result
+    ...    and set RTE relay to OFF state. Implementation must be
+    ...    compatible with the theory of operation of a specific
+    ...    platform.
+    Sleep    1s
+    Power Cycle Off
+    IF    '${FLASH_OPT}'=='full'
```
[the remark this answers is not in the document]
Comment (PLangowski) [reply platform-configs/minnowboard-turbot.robot:55]: My bad, I took this from the old config you linked and forgot to check if this variable exists here. It doesn't. Which of the 2 flashrom commands do you think should be used here? The second one reads the IFD, but I don't know if we need that here.

[the remark this answers is not in the document]
Comment (macpijan) [reply platform-configs/minnowboard-turbot.robot:55]: The one flashing full chip, with no IFD read

File: platform-configs/minnowboard-turbot.robot [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        ${flash_result}    ${rc}=    SSHLibrary.Execute Command
+        ...    flashrom -f -p linux_spi:dev=/dev/spidev1.0,spispeed=8000 -c W25Q64JV-.Q -w /tmp/coreboot.rom 2>&1
+        ...    return_rc=True
+    ELSE
+        ${flash_result}    ${rc}=    SSHLibrary.Execute Command
+        ...    flashrom -f -p linux_spi:dev=/dev/spidev1.0,spispeed=8000 -w /tmp/coreboot.rom --ifd -i bios 2>&1
```
[the remark this answers is not in the document]
Comment (PLangowski) [reply platform-configs/minnowboard-turbot.robot]: Done 6bd5d5d5c596af59a23ac3e48cf452ed7e1a7990

File: platform-configs/minnowboard-turbot.robot (hunk 3 above)
[the remark this answers is not in the document]
Comment (PLangowski) [reply platform-configs/minnowboard-turbot.robot]: Done 6bd5d5d5c596af59a23ac3e48cf452ed7e1a7990

**Review by macpijan — APPROVED** [review]

**Pull request merged**